ports/gdal/dependency_win.cmake

```
@@ -234,4 +234,19 @@ macro(find_dependency_win)
     endif()
     set(HAVE_SPATIALITE "-DHAVE_SPATIALITE")
   endif()
+
+  # Setup hdf5 libraries
+  # Not supported on WIN32 platforms with static linking
+  if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    if(EXISTS "${CURRENT_INSTALLED_DIR}/lib/hdf5.lib")

```

Since `hdf5` only be actived when using feature `netcdf-c`, I think we should replace it with:
```suggestion
    if("netcdf-c" IN_LIST FEATURES)
```